nix/haskell-dependencies.nix

```diff
@@ -1,49 +0,0 @@
-# Extract our Haskell dependencies from the set of all haskell packages in
```

Comment:
Why remove this file? It makes `hoff.nix` somehow chunkier.

Reply:
> Why remove this file?

It used to be a patch to get (Stack and HLS) to cooperate while inside Nix environments. Now that Stack is out of the way, we don't need it anymore and can define Nix derivations as in Nixpkgs. One not-quite-hacky-but-not-extensible solution would be `cabal2nix > hoff.nix` and not touching it for a while, but it's slowly pushed out of the way as the Haskell Nixpkgs tooling improves.